Initial version of the Ontology for Taxon Concepts and Names

hlapp opened the pull request:
Quoting from the description on the ontology:

> The Ontology for Taxon Concepts And Names (TCAN) is an application ontology for expressing, matching, and resolving taxon concepts and their names as encountered in the wild, such as in publications, taxon mentions, clade definitions, etc. One of the key use cases for which it is being developed is for expressing and matching specifiers (in essence taxon concepts) used in clade definitions.
>
> The TCAN ontology is a synthesis of a variety of relevant vocabularies and ontologies, including in particular the TDWG TaxonName, TaxonConcept, and DarwinCore vocabularies; the NOMEN ontology (SpeciesFileGroup); the OpenBiodiv Ontology (Pensoft); and the Comparative Data Analysis Ontology (CDAO).
>
> One of the express design goals of TCAN is to avoid as much as possible creating new terms, whether classes or properties. Hence, there will be few if any terms here that are not already present in its source vocabularies. Instead, TCAN considers its added value in the following.
> 1. Provide an ontology for its application domain that stands on its own, meaning all necessary axioms are either directly included, or are imported from locations that resolve automatically and reliably.
> 2. Select subsets suitable for its application domain from each source vocabulary, and drop extraneous axioms and annotations.
> 3. Apply error corrections and replace non-standard annotation properties with standard ones. Ensure there are no parse nor reasoner errors.
> 4. Supplement subclass, type, class expression and other axioms to promote alignment and interoperability between the source ontologies.
> 5. Ensure expressity is at least within the OWL-DL, and ideally within the OWL-EL profile.

I came to the need for this ontology in the course of trying to rethink and revamp how we express phyloreferences. I had to realize that the space of already existing vocabularies and ontologies for taxon concepts and taxon names is kind of a mess, with IRIs that don't resolve by themselves, datatype errors, reasoner errors due to missing declarations, spelling errors, and most possible cross-links missing, to name just a few.

Comments welcome on this synthesis.

Closes #10.

hlapp committed:
Initial version of the Ontology for Taxon Concepts and Names

hlapp committed:
Fixes default namespace incorrectly by Protégé

For some reason Protégé uses the default namespace for abbreviating the
IRI for rdfs:isDefinedBy annotations, but then defines it incorrectly as
the IRI for the ontology.

hlapp committed:
Fixes typo in metadata

hlapp committed:
Adds a few DwC and TC properties

hlapp committed:
Places tc:hasName under dwciri:scientificName

hlapp commented:
@mjy just FYI, in case you are curious. I'm reusing a lot of NOMEN terms here, and putting some class expression axioms on them to promote interoperability.

hlapp committed:
Removes domain constraint on tn:nomenclaturalCode

Otherwise, with a domain constraint to tn:TaxonName, we can't use it for
rank terms, or otherwise we'd infer that rank terms are TaxonName instances,
which would be false.

mjy commented:
@hlapp Thanks very much for the ping (and feedback on NOMEN).  Keep poking at me, I need to sit down and do some hard grokking as to how this all integrated.

hlapp committed:
Corrects spelling of Bacterial nomenclator

ICNB changed its name to ICNP, see SpeciesFileGroup/nomen#13.

hlapp committed:
Re-labels dwc:scientificName to 'Has Scientific Name'

This is so it doesn't conflict for authoring in Protégé with an
OpenBiodiv-O class. The label in the [DWC RDF] is 'Scientific Name'.
Because only the label is being changed, there should be no effect on
semantics.

hlapp committed:
Assigns initial versionIRI

hlapp committed:
Generates TCAN in OWL format and deploys to web

hlapp commented:
I'm closing the comment period on this PR tonight. Further comments continue to be most welcome, simply post on issue tracker.
